github/teams.go

```diff
@@ -121,6 +121,25 @@ func (s *TeamsService) GetTeam(ctx context.Context, team int64) (*Team, *Respons
 	return t, resp, nil
 }
 
+// GetTeamByName fetches a team by slug.
```

Comment:
Since the parameter is called `slug` and your documentation also says `slug`, and the GitHub docs say "the team's slug", what do you think about calling this method `GetTeamBySlug` ?
I'm actually fine either way, but I had no clue what a slug was until I looked at the linked GitHub docs.

Reply:
It was just to use the same name used as title inside the github documentation.
I'm fine changing the name of the method to `GetTeamBySlug` if needed.

Reply:
Right... that makes total sense. Let's see what @gauntface thinks.

Reply:
Sorry for the delay on this - GetTeamBySlug please.

The docs say: `/orgs/:org/teams/:team_slug` so I would definitely use slug, since it's different from the name.

Reply:
@gauntface, @gmlewis I renamed the method. Thank you for your review.